minmea.c

```diff
@@ -401,6 +401,38 @@ bool minmea_parse_gbs(struct minmea_sentence_gbs *frame, const char *sentence)
     return true;
 }
 
+bool minmea_parse_gns(struct minmea_sentence_gns *frame, const char *sentence)
+{
+    // $GNGNS,103600.01,5114.51176,N,00012.29380,W,ANNN,07,1.18,111.5,45.6,,,V*00
+    char type[6];
+    // NOTE: NMEA revealed says the posMode field can be 1-4 chars, however, some receivers have more (ublox MAXM10S returns 6)
+    static char value[10]; // Altenative is pass a buffer to hold this, instead of declaring as static
```

Comment:
declaring this "static" means that the buffer will be allocated once, in the .bss, and will be available within the scope of this file.

Instead, change the struct you created to have `char posMode[10]` and then replace the `value` argument in the call to minmea_scan with `frame->postMode,`.